plugin/src/main/java/org/opensearch/ml/action/upload_chunk/MLModelMetaCreate.java

```diff
@@ -0,0 +1,90 @@
+import org.opensearch.ml.common.transport.upload_chunk.MLCreateModelMetaInput;
+import org.opensearch.ml.indices.MLIndicesHandler;
+import org.opensearch.threadpool.ThreadPool;
+
+@Log4j2
+public class MLModelMetaCreate {
+
+    private final MLIndicesHandler mlIndicesHandler;
+    private final ThreadPool threadPool;
+    private final Client client;
+
+    @Inject
+    public MLModelMetaCreate(MLIndicesHandler mlIndicesHandler, ThreadPool threadPool, Client client) {
+        this.mlIndicesHandler = mlIndicesHandler;
+        this.threadPool = threadPool;
+        this.client = client;
+    }
+
+    public void createModelMeta(MLCreateModelMetaInput mlCreateModelMetaInput, ActionListener<String> listener) {
+        try {
+            String modelName = mlCreateModelMetaInput.getName();
+            String version = mlCreateModelMetaInput.getVersion();
+            FunctionName functionName = mlCreateModelMetaInput.getFunctionName();
+            try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+                mlIndicesHandler.initModelIndexIfAbsent(ActionListener.wrap(res -> {
+                    MLModel mlModelMeta = MLModel
+                        .builder()
+                        .name(modelName)
+                        .algorithm(functionName)
+                        .version(version)
+                        .modelFormat(mlCreateModelMetaInput.getModelFormat())
+                        .modelState(MLModelState.UPLOADING)
+                        .modelConfig(mlCreateModelMetaInput.getModelConfig())
+                        .totalChunks(mlCreateModelMetaInput.getTotalChunks())
+                        .modelContentHash(mlCreateModelMetaInput.getModelContentHashValue())
+                        .modelContentSizeInBytes(mlCreateModelMetaInput.getModelContentSizeInBytes())
+                        .createdTime(Instant.now())
+                        .build();
+                    IndexRequest indexRequest = new IndexRequest(ML_MODEL_INDEX);
+                    indexRequest
+                        .source(mlModelMeta.toXContent(XContentBuilder.builder(XContentType.JSON.xContent()), ToXContent.EMPTY_PARAMS));
+                    indexRequest.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+
+                    client.index(indexRequest, ActionListener.wrap(r -> {
+                        log.debug("Index model meta doc successfully {}", modelName);
+                        listener.onResponse(r.getId());
+                    }, e -> {
+                        log.error("Failed to index model meta doc", e);
+                        listener.onFailure(e);
+                    }));
+                }, ex -> {
+                    log.error("Failed to init model index", ex);
+                    listener.onFailure(ex);
+                }));
+            } catch (Exception e) {
+                log.error("Failed to create model meta doc", e);
```

Comment:
what does `Failed to init ` mean? Any difference from `Failed to create`? If not, Can we be consistent wth: `Failed to create ` which is easy to understand. Can please reflect it where it is applicable? IMO `Failed to init` or `Failed to index` is bit confusing in this context.